src/components/screens/timebank-view-all-screen.tsx

```diff
@@ -0,0 +1,290 @@
+              <Search />
+            </InputAdornment>
+          )
+        }}
+      />
+    </Box>
+  );
+
+  /**
+   * Handle search input change
+   *
+   * @param event input change event
+   */
+  const handleSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newSearchInput = event.target.value;
+    setSearchInput(newSearchInput);
+
+    if (newSearchInput === "") {
+      setDisplayedPersonsTotalTime(personsWithTotalTime);
+      return;
+    }
+
+    const newDisplayedPersonsTotalTime = personsWithTotalTime.filter((personsWithTotalTime) =>
+      `${personsWithTotalTime.person.firstName} ${personsWithTotalTime.person.lastName}`
+        .toLowerCase()
+        .includes(newSearchInput.toLowerCase())
+    );
+
+    setDisplayedPersonsTotalTime(newDisplayedPersonsTotalTime);
+  };
+
+  /**
+   * Get balance color
+   *
+   * @param personTotalTime person total time object
+   * @returns color string
+   */
+  const getBalanceColor = (personTotalTime: PersonTotalTime) =>
+    personTotalTime.balance > 0 ? theme.palette.success.main : theme.palette.error.main;
+
+  return (
+    <>
+      <Card sx={{ marginBottom: 2 }}>{renderSearch()}</Card>
+      <Grid container spacing={2} textAlign={"center"} marginBottom={20}>
+        {loading ? (
+          <CircularProgress sx={{ margin: "auto", mt: "5%", mb: "5%" }} />
+        ) : (
+          displayedPersonsTotalTime.map((personWithTotalTime) => {
+            return (
+              <Grid item xs={12} sm={6} md={4} key={personWithTotalTime.person.id}>
+                <Card
+                  sx={{
+                    display: "flex",
+                    flexDirection: "column",
+                    alignItems: "center",
+                    height: "400px",
+                    "& .recharts-wrapper .recharts-surface": {
+                      overflow: "visible"
+                    }
+                  }}
+                >
+                  <Box sx={{ padding: "10px", marginBottom: "60px" }}>
+                    <Typography variant="h4">{`${personWithTotalTime.person.firstName} ${personWithTotalTime.person.lastName}`}</Typography>
+                    <Typography>
+                      {personWithTotalTime.person.startDate &&
+                        DateTime.fromFormat(personWithTotalTime.person.startDate, "yyyy-M-d")
+                          .setLocale(language)
+                          .toLocaleString()}
+                    </Typography>
+                  </Box>
+                  {renderPieChart(personWithTotalTime, false)}
+                  <Box sx={{ marginTop: "60px" }}>
+                    {personWithTotalTime.personTotalTime && (
+                      <Typography
+                        fontSize={22}
+                        color={getBalanceColor(personWithTotalTime.personTotalTime)}
+                      >
+                        {getHoursAndMinutes(personWithTotalTime.personTotalTime.balance)}
```

Comment:
Again maybe a string or tooltip here to say balance:

Reply:
Added a string here to say "Balance: "